packages/venia-concept/src/actions/checkout.js

```diff
@@ -1,80 +1,101 @@
+import { createActions } from 'redux-actions';
 import { RestApi } from '@magento/peregrine';
 
 import { closeDrawer } from 'src/actions/app';
-import { getGuestCartId } from 'src/actions/cart';
-import * as durations from 'src/shared/durations';
-import timeout from 'src/util/timeout';
+import { clearGuestCartId, getGuestCartId } from 'src/actions/cart';
 
-const { request } = RestApi.Magento2;
+const prefix = 'CHECKOUT';
+const actionTypes = ['EDIT', 'RESET'];
 
-const mockAddress = {
-    country_id: 'US',
-    firstname: 'Veronica',
-    lastname: 'Costello',
-    street: ['6146 Honey Bluff Parkway'],
-    city: 'Calder',
-    postcode: '49628-7978',
-    region_id: 33,
-    region_code: 'MI',
-    region: 'Michigan',
-    telephone: '(555) 229-3326',
-    email: 'veronica@example.com'
+// classify action creators by domain
+// e.g., `actions.order.submit` => CHECKOUT/ORDER/SUBMIT
+// a `null` value corresponds to the default creator function
+const actionMap = {
+    CART: {
+        SUBMIT: null,
+        ACCEPT: null,
+        REJECT: null
+    },
+    INPUT: {
+        SUBMIT: null,
+        ACCEPT: null,
+        REJECT: null
+    },
+    ORDER: {
+        SUBMIT: null,
+        ACCEPT: null,
+        REJECT: null
+    }
 };
 
-const enterSubflow = () =>
+const actions = createActions(actionMap, ...actionTypes, { prefix });
+export default actions;
```

Comment:
`createActions`, provided by `redux-actions`, has a [wonderful, optional API](https://redux-actions.js.org/api/createaction#createactionsactionmap-identityactions) that allows us to nest action creators. While it's a bit overkill for most action types, I think it's perfect for action types associated with API requests, which often fit a request-success-error pattern.

Reply:
Yeah, let's go with prior art here. There should be simple recipes for the sequence of action dispatch API calls and other common interactions (say, additional logic loaded via a chunk?), like:

```js
import { lifecycle } from '@magento/peregrine';

const cartLifecycle = lifecycle('CART');

const cartApiLifecycle = cartLifecycle.requests({
    GET_DETAILS(state) {
       return {
         path: `/guest-carts/${state.guestCartId}`
      }
    },
    ADD_ITEM(state, item) {
       return {
          path: `/guest-carts/${state.guestCartId}/items/`,
          method: 'POST',
          body: item
      }
  },
  async (path, method, body) => M2ApiRequest.request(path, { method, body })
);

const [getDetails, addItem] = cartApiLifecycle.getActions();

const reducer = cartApiLifecycle.getReducer();

```

We don't need to make those abstractions yet, of course.